### PR TITLE
update-users and delete-users

### PR DESCRIPTION
**Pull request by Gahamanyiced**
What this PR do?
- Update a user info
- Delete a user

### CODE REVIEW

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -167,11 +168,65 @@ export class userControllers {
...
+          phone,
+          email,
+          country,
+          roleId,
+          driving_licence,
+        },
+        user
+      );
+      return res.status(200).json({
+        message: 'User updated successful!',
```
Comment (TuyizeeAnastase) [inline src/controllers/user.controller.js:204]: successfully

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -167,11 +168,65 @@ export class userControllers {
...
+      return res.status(500).json({
+        message: 'Error while updating a user',
+        error: error.message,
+      });
+    }
+  }
+  async deleteUser(req, res) {
+    try {
+      const user = req.user;
+      await deleteUsers(user.id);
```
Comment (TuyizeeAnastase) [inline src/controllers/user.controller.js:217]: change deleteUsers function to deleteUser

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -167,11 +168,65 @@ export class userControllers {
...
+        user: { id: user.id, ...req.body },
+      });
+    } catch (error) {
+      return res.status(500).json({
+        message: 'Error while updating a user',
+        error: error.message,
+      });
+    }
+  }
+  async deleteUser(req, res) {
```
Comment (TuyizeeAnastase) [inline src/controllers/user.controller.js:214]: change deleteUser to userDelete

File: src/controllers/user.controller.js [hunk 4]
```diff
@@ -167,11 +168,65 @@ export class userControllers {
...
     } catch (error) {
       return res.status(500).json({
-        message: "Error while getting users",
+        message: 'Error while getting users',
         error: error.message,
       });
     }
   }
+
+  async updateUser(req, res) {
```
Comment (TuyizeeAnastase) [inline src/controllers/user.controller.js:177]: change updateUser to userUpdate, it is confusing with other function at the bottom `await updateUser`

**Review by TuyizeeAnastase — APPROVED** [review]
Improve Pull request description and how to test 
refer to this https://github.com/Motion-CodeHouse-Ltd/backend-kwaco/pull/14

**Pull request merged**